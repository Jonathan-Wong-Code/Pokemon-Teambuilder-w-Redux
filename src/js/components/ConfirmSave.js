import React from 'react';

export class ConfirmSave extends React.Component {
  constructor() {
    super();

    this.closeBtn = React.createRef();
  }

  componentDidMount() {
    this.closeBtn.current.focus();
  }
  
  render() {
    return (
      <div 
        className='modal__background' 
        onClick={this.props.handleToggleSaveModal} 
        data-test='confirm-save-component'
      >
        <div className='confirm-save__modal' onClick={(e) => e.stopPropagation()}
          data-test='confirm-save-body'
        >
          <p data-test='confirm-save-text'>Saved <span className='confirm-save__name'>
              {this.props.pokemon.name}
            </span>
          </p>
          <button 
            onClick={this.props.handleToggleSaveModal} 
            className='btn confirm-save__btn'
            data-test='confirm-save-close-btn'
            ref={this.closeBtn}
          >
          Close
          </button>
        </div>
      </div>
    );
  }
}

export default ConfirmSave;
