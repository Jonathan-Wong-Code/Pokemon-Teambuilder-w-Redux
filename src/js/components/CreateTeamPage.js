import React from 'react';
import { connect } from 'react-redux';
import BuildPokeTeamPage from './BuildPokeTeamPage';
import { addPokeTeam } from './../actions/pokeTeam';

export const CreateTeamPage = ({ history, addPokeTeam }) => {
  const handleAddTeam = (pokeTeam) => {
    addPokeTeam(pokeTeam);
  };
 
  return (
    <BuildPokeTeamPage 
      type='create' 
      handleAddTeam={handleAddTeam}
      history={history}
      data-test='build-team-component'
    />
  );
};

const mapDispatchToProps = (dispatch) => ({
  addPokeTeam : (pokeTeam) => dispatch(addPokeTeam(pokeTeam))
});

export default connect(null, mapDispatchToProps)(CreateTeamPage);
