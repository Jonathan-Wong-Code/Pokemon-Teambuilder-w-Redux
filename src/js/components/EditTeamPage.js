import React from 'react';
import { connect } from 'react-redux';
import BuildPokeTeamPage from './BuildPokeTeamPage';
import { editPokeTeam } from './../actions/pokeTeam';

export const EditTeamPage = ({ editPokeTeam, editedTeam, history }) => {
  const handleEditTeam = (pokeTeam) => {
    editPokeTeam(pokeTeam);
  };

  return (
    <BuildPokeTeamPage 
      type='edit' 
      handleEditTeam={handleEditTeam} 
      pokeTeam={editedTeam}
      history={history}
      data-test='build-page-component'
    />
  );
};

const mapDispatchToProps = (dispatch) => ({
  editPokeTeam : (pokeTeam) => dispatch(editPokeTeam(pokeTeam))
});

const mapStateToProps = (state, props) => ({
  editedTeam : state.savedTeams[props.match.params.id]
});

export default connect(mapStateToProps, mapDispatchToProps)(EditTeamPage);
