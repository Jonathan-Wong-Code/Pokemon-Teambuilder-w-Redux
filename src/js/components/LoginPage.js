import React from 'react';
import { connect } from 'react-redux';
import { startLogin, loginGuest } from './../actions/auth';


export const LoginPage = ({ startLogin, loginGuest, history }) => {
  const onLoginClick = () => {
    startLogin();
  };

  const onGuestLoginClick = () => {
    loginGuest();
    history.push('/create');
  };
  
  return (
    <section className='login' data-test='login-component'>
      <div className='login__buttons'>
        <button 
          onClick={onLoginClick} 
          className='login__button btn'
          data-test='login-button'
        >Login with Google
        </button>
        <button 
          onClick={onGuestLoginClick} 
          className='login__button btn'
          data-test='login-guest-button'
        >Guest
        </button>
      </div>
    </section>
  );
};

const mapDispatchToProps = (dispatch) => ({
  startLogin : () => dispatch(startLogin()),
  loginGuest : () => dispatch(loginGuest())
});

export default connect(null, mapDispatchToProps)(LoginPage);
