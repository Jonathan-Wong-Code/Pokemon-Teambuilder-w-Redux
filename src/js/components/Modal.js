import React from 'react';
import PokemonTypes from './PokemonTypes';

export class Modal extends React.Component{
  constructor(){
    super();

    this.pokeModalButton = React.createRef();
  }

  componentDidMount (){
    this.pokeModalButton.current.focus();
  }
  

  renderTypes = () => this.props.pokemon.types.map(type => {
    return <PokemonTypes type={type} key={type.type.name} />;
  });
  
  renderAbilities = () => this.props.pokemon.abilities.map(ability => {
    return (
      <li 
        className='modal__ability' 
        key={ability.ability.name}
        data-test='modal-ability'
      > 
        {ability.ability.name} 
      </li>    
    );
  });

  renderStats = () => this.props.pokemon.stats.map(stat => (
    <li key={stat.stat.name} data-test='modal-stat'>
      <span className='modal__stat-name'>
        { stat.stat.name }:
      </span> {stat.base_stat}
    </li>
  ));

  render() {
    return (
      <div className='modal__background' onClick={this.props.handleTogglePokeModal}
        data-test='modal-component'
      >
        <div className='modal__body' onClick={(e) => e.stopPropagation()}
          data-test='modal-body'
        >
          <div className='modal__img-box'>
            <img 
              src={this.props.pokemon.sprites.front_default} 
              alt={this.props.pokemon.name} 
              className='modal__img'
              data-test='modal-img'
            />
          </div>
          <h2 className='modal__name' data-test='modal-name'>
            {this.props.pokemon.name}
         </h2>
          <ul className='modal__attributes'>
            <li className='modal__types' data-test='modal-type'>
              <span className='modal__stat-name'>type: </span>
              {this.renderTypes()}
            </li>
      
            <li className='modal__stats'>
              <ul className='modal__stats-list'>
                {this.renderStats()}
              </ul>
            </li>
  
            <li className='modal__abilities'>
              <span className='modal__stat-name modal__ability-tag'>abilities: </span>
              <ul className='modal__abilities-list'>
                {this.renderAbilities()}
              </ul>
            </li>
          </ul>
          {this.props.error && <p className='modal__error'>{this.props.error}</p>}
          <div className='modal__buttons'>
            <button 
              onClick={this.props.handleAddPokemon} 
              className='btn modal__btn'
              ref={this.pokeModalButton}
              data-test='modal-save'
            >
               Save to team
            </button>
            <button 
              onClick={this.props.handleTogglePokeModal} 
              className='btn modal__btn'
              data-test='modal-cancel'
            >
               Cancel
            </button>
          </div>
        </div>
      </div>   
    );
  }
};

export default Modal;
