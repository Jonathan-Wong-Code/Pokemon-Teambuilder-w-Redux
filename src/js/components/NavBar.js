import React from 'react';
import { NavLink } from 'react-router-dom';
import { connect } from 'react-redux';
import { startLogout, logout } from './../actions/auth';

export const NavBar = ({ startLogout, history, auth, logout }) => {
  const onLogoutClick = () => {
    if (auth === 'guest') {
      logout();
      history.push('/');
    } else {
      startLogout();
    }
  };

  return (
    <nav className='navbar' data-test='nav-bar-component'>
      <ul className='navbar__nav'>
        <li className='navbar__item'>
          <NavLink to='/create' activeClassName='is-active' className='navbar__link btn'>
            Create Team
          </NavLink>
        </li>
        <li className='navbar__item'>
          <NavLink to='/SavedTeams' activeClassName='is-active' className='navbar__link btn'>
            Saved Teams
          </NavLink>
        </li>
        <li className='navbar__item'>
          <button 
            onClick={onLogoutClick} 
            className='btn navbar__logout'  
            data-test='nav-bar-logout'>
            Logout
          </button>
        </li>
      </ul>
    </nav>
  );  
};

const mapStateToProps = (state) => ({
  auth : state.auth.uid
});

const mapDispatchToProps = (dispatch) => ({
  startLogout : () => dispatch(startLogout()),
  logout : () => dispatch(logout())
});

export default connect(mapStateToProps, mapDispatchToProps)(NavBar);
