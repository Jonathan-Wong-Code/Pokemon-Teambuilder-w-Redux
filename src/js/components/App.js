import React from 'react';
import { Router, Switch } from 'react-router-dom';
import createHistory from 'history/createBrowserHistory';
import CreateTeamPage from './CreateTeamPage';
import SavedTeamsPage from './SavedTeamsPage';
import EditTeamPage from './EditTeamPage';
import LoginPage from './LoginPage';
import PrivateRoute from './../routes/PrivateRoute';
import PublicRoute from './../routes/PublicRoute';

const history = createHistory();

const App = () => {
  return (
    <Router history={history}>
      <React.Fragment>
        <Switch>
          <PublicRoute 
            path='/' 
            component={LoginPage} 
            exact
          />

          <PrivateRoute 
            path='/create' 
            component={CreateTeamPage} 
          />
          <PrivateRoute
            path='/SavedTeams' 
            component={SavedTeamsPage} 
          />
          <PrivateRoute 
            path='/edit/:id' 
            component={EditTeamPage} 
          />
        </Switch>
      </React.Fragment>
    </Router>
  );
};

export default App;
