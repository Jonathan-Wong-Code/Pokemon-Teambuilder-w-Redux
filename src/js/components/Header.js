import React from 'react';
import { connect } from 'react-redux';
import NavBar from './NavBar';

import togepi from './../../images/togepi.png';

// const togepi = require('./../../images/togepi.png');
const pikachu = require('./../../images/pikachu.png');
const squirtle = require('./../../images/squirtle.png');
const logo = require('./../../images/logo.png');
const bulbasaur = require('./../../images/bulbasaur.png');
const charmander = require('./../../images/charmander.png');
const eevee = require('./../../images/eevee.png');

export const Header = ({ history, isAuthenticated }) => {
  const headerClasses = isAuthenticated ? 'header' : 'header header--login';
  const headingClasses = isAuthenticated ? 'header__heading' : 'header__heading header__heading--login';
  const pokeBannerClasses = isAuthenticated ? 'poke-banner' : 'poke-banner poke-banner--login';

  return (
    <header className={headerClasses} data-test='header-component'> 
      <ul className={pokeBannerClasses}>
        <li className='poke-banner__item'>
          <img src={togepi} alt='togepi' className='poke-banner__img' />
        </li>

        <li className='poke-banner__item'>
          <img src={pikachu} alt='pikachu' className='poke-banner__img' />
        </li>

        <li className='poke-banner__item'>
          <img src={squirtle} alt='squirtle' className='poke-banner__img' />
        </li>

        <li className='poke-banner__item poke-banner__logo'>
          <img src={logo} alt='pokemon logo' className='poke-banner__img' />
        </li>

        <li className='poke-banner__item'>
          <img src={eevee} alt='eevee' className='poke-banner__img' />
        </li>

        <li className='poke-banner__item'>
          <img src={charmander} alt='charmander' className='poke-banner__img' />
        </li>
        
        <li className='poke-banner__item'>
          <img src={bulbasaur} alt='bulbasaur' className='poke-banner__img' />
        </li>
      </ul>
      <h1 className={headingClasses}>Poke-Team Picker</h1>
      {isAuthenticated && <NavBar history={history} />}
    </header>
  );
};

const mapStateToProps = (state) => ({
  isAuthenticated : !!state.auth.uid
});

export default connect(mapStateToProps)(Header);
