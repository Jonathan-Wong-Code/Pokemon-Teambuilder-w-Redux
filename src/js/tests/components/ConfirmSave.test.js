
import React from 'react';
import { mount } from 'enzyme';
import pokemon from './../fixtures/pokemon';
import { ConfirmSave } from './../../components/ConfirmSave';

describe("The unconnected ConfirmSave component", () => {
  let wrapper, handleToggleSaveModal;

  beforeEach(() => {
    handleToggleSaveModal = jest.fn();
    const defaultProps = {
      handleToggleSaveModal,
      pokemon
    };
    wrapper = mount(<ConfirmSave {...defaultProps} />);
  });

  afterEach(() => {
    wrapper.unmount();
  });

  it('Should render a snapshot of the component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('Should render the ConfirmSave component', () => {
    expect(wrapper.find(`[data-test='confirm-save-component']`).length).toBe(1);
  });

  it('Should render the correct text in the box', () => {
    expect(wrapper.find(`[data-test='confirm-save-text']`).text()).toEqual(`Saved ${pokemon.name}`);
  });

  it('Should close the modal when you click on the background', () => {
    wrapper.find(`[data-test='confirm-save-component']`).simulate('click');
    expect(handleToggleSaveModal).toHaveBeenCalled();
  });

  it('Should close the modal when you click on the close button', () => {
    wrapper.find(`[data-test='confirm-save-close-btn']`).simulate('click');
    expect(handleToggleSaveModal).toHaveBeenCalled();
  });

  it('Should not close the modal when you click on the body', () => {
    wrapper.find(`[data-test='confirm-save-body']`).simulate('click');
    expect(handleToggleSaveModal).not.toHaveBeenCalled();
  });
});
