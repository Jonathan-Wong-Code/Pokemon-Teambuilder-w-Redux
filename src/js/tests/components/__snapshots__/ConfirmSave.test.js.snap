// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The unconnected ConfirmSave component Should render a snapshot of the component 1`] = `
<ConfirmSave
  handleToggleSaveModal={[MockFunction]}
  pokemon={
    Object {
      "abilities": Array [
        Object {
          "ability": Object {
            "name": "ability1",
          },
        },
        Object {
          "ability": Object {
            "name": "ability2",
          },
        },
      ],
      "name": "poketest",
      "sprites": Object {
        "front_default": "sprite",
      },
      "stats": Array [
        Object {
          "base_stat": 1,
          "stat": Object {
            "name": "stat1",
          },
        },
        Object {
          "base_stat": 2,
          "stat": Object {
            "name": "stat2",
          },
        },
        Object {
          "base_stat": 3,
          "stat": Object {
            "name": "stat3",
          },
        },
        Object {
          "base_stat": 4,
          "stat": Object {
            "name": "stat4",
          },
        },
        Object {
          "base_stat": 5,
          "stat": Object {
            "name": "stat5",
          },
        },
        Object {
          "base_stat": 6,
          "stat": Object {
            "name": "stat6",
          },
        },
      ],
      "types": Array [
        Object {
          "type": Object {
            "name": "type",
          },
        },
      ],
      "uniqueId": "pokeId",
      "url": "testabc123",
    }
  }
>
  <div
    className="modal__background"
    data-test="confirm-save-component"
    onClick={[MockFunction]}
  >
    <div
      className="confirm-save__modal"
      data-test="confirm-save-body"
      onClick={[Function]}
    >
      <p
        data-test="confirm-save-text"
      >
        Saved 
        <span
          className="confirm-save__name"
        >
          poketest
        </span>
      </p>
      <button
        className="btn confirm-save__btn"
        data-test="confirm-save-close-btn"
        onClick={[MockFunction]}
      >
        Close
      </button>
    </div>
  </div>
</ConfirmSave>
`;
